commit 76a6a153dba7fcb34088b713663c81fbabdb7d98 (refs/changes/81/1481/1)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2017-02-09T13:37:23-05:00 (2 years, 8 months ago)
    
    AGENT-1060 test.VmAgentRealVmadm fails to parse output from ptree(1) reliably

diff --git a/tests/test.VmAgentRealVmadm.js b/tests/test.VmAgentRealVmadm.js
index fef81ba..253410a 100644
--- a/tests/test.VmAgentRealVmadm.js
+++ b/tests/test.VmAgentRealVmadm.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright 2017, Joyent, Inc.
  */
 
 var execFile = require('child_process').execFile;
@@ -1092,13 +1092,16 @@ test('VmAgent works after initial errors', function _test(t) {
 
                     if (!err) {
                         stdout.split('\n').forEach(function _onLine(line) {
-                            var proc = line.trim().split(' ');
+                            var proc = line.trim().match(/(\d+)\s+(.*)$/);
 
-                            if (proc[1]
-                                && proc[1] === '/usr/vm/sbin/zoneevent') {
+                            /* eslint-disable no-magic-numbers */
+                            if (proc && proc[2]
+                                && proc[2] === '/usr/vm/sbin/zoneevent') {
                                 // track PIDs so we can kill them
-                                zoneevent_children.push(proc[0]);
+                                zoneevent_children.push(proc[1]);
                             }
+
+                            /* eslint-enable no-magic-numbers */
                         });
 
                         t.equal(zoneevent_children.length, 1,
